### PR TITLE
docs: cabal.project in parent directories

### PR DESCRIPTION
**Pull request by ulysses4ever**
This is a stop-gap measure against #7930. It's probably not ideal so I'm fine with just waiting for a better feature, if you think it's the right way forward.

### CODE REVIEW

**Review by Mikolaj — APPROVED** [review]
That's definitely progress.

File: doc/cabal-project.rst [hunk 1]
```diff
@@ -32,6 +32,8 @@ options):
 
 4. ``cabal.project.local`` (the output of ``cabal v2-configure``)
 
+A call to ``cabal build`` will consider ``cabal.project*`` from parent
```
Comment (andreasabel) [inline doc/cabal-project.rst:35]: Just some wording suggestion:
_Any_ call to ... will consider ... _files_ from parent ...

**Review by andreasabel — COMMENTED** [review]
I'd like to point out that _parent directory_ is ambiguous when symlinks are involved.
If I have a directory `a/d` and a link `b/d -> a/d` and I am sitting in `b/d` then both `a` and `b` can be parent directories depending on whether the current directory is canonicalized (becoming `a/d` or not staying `b/d`).  Also _going up_ is ambiguous and OS-specific when symlinks are involved.  
(We discussed this in Agda in the context of project files, but atm I cannot find the discussion in the issues/PRs...)

I don't think this PR should be blocked because of this ambiguity, but an issue should be opened to point out the problem with a TODO that the documentation needs to be clarified.

File: doc/cabal-project.rst (hunk 1 above)
Comment (ulysses4ever) [reply doc/cabal-project.rst]: Done, thanks!

**Review by Kleidukos — APPROVED** [review]

**Pull request merged**